packages/origin/contracts/Origin/CertificateLogic.sol

```
@@ -158,6 +158,39 @@ contract CertificateLogic is CertificateInterface, CertificateSpecificContract,
         }
     }
 
+    /// @notice claims a set of certificates
+    /// @param _idArray the ids of the certificates to be claimed
+    function claimCertificateBulk(uint[] calldata _idArray) external {
+        for (uint i = 0; i < _idArray.length; i++) {
+            retireCertificateInternal(_idArray[i], msg.sender);
+        }
+    }
+
+    /// @notice Request a certificate to retire. Only Certificate owner can retire
+    /// @param _certificateId The id of the certificate
+    function retireCertificate(
+        uint _certificateId
+    )
+        external
+    {
+        retireCertificateInternal(_certificateId, msg.sender);
+    }
+
+    function retireCertificateInternal(uint _certificateId, address claimer) internal {
+        CertificateDB.Certificate memory cert = CertificateDB(address(db)).getCertificate(_certificateId);
+        require(cert.tradableEntity.owner == claimer, "You have to be the owner of the contract.");
+        require(
+            cert.certificateSpecific.children.length == 0,
+            "Unable to retire certificates that have already been split."
+        );
+
+        if (cert.certificateSpecific.status != uint(CertificateSpecificContract.Status.Retired)) {
```

Would it make more sense to replace this with `require(cert.certificateSpecific.status != uint(CertificateSpecificContract.Status.Retired), "cannot retired a certificate that has already been retired");`? So that it doesn't silently fail.